scripts/FigS2B_adt-plots.R

```diff
@@ -0,0 +1,143 @@
+
+library(ggplot2)
+library(SingleCellExperiment)
+
+theme_set(
+  theme_classic() +
+    theme(
+      strip.background = element_rect(fill = "transparent"),
+      # no axis ticks or labels
+      axis.line = element_blank(),
+      axis.ticks = element_blank(),
+      axis.text = element_blank(),
+      # add a square around each of the plots
+      panel.background = element_rect(colour = "black", linewidth=0.5),
+      aspect.ratio = 1
+    )
+)
+
+set.seed(2024)
+
+
+# Set up -----------------------------------------------------------------------
+
+# define file paths and read in sce objects
+data_dir <- here::here("s3_files", "SCPCS000216")
+
+filtered_sce_file <- file.path(data_dir, "SCPCL000290_filtered.rds")
+filtered_sce <- readr::read_rds(filtered_sce_file)
+
+processed_sce_file <- file.path(data_dir, "SCPCL000290_processed.rds")
+processed_sce <- readr::read_rds(processed_sce_file)
+
+output_plot_file <- here::here("figures", "pngs", "FigS2B_adt-plots.png")
+
+# Filtered ADT plot ------------------------------------------------------------
+
+# grab coldata from filterd object 
+filtered_coldata_df <- colData(filtered_sce) |>
+  as.data.frame() |>
+  tibble::rownames_to_column("barcode")
+
+filter_levels <- c("Keep", "Filter (RNA & ADT)", "Filter (RNA only)", "Filter (ADT only)")
+filtered_coldata_df <- filtered_coldata_df |>
+  # add column to represent filtering on both RNA and ADT counts
+  dplyr::mutate(filter_summary = dplyr::case_when(
+    scpca_filter == "Keep" & adt_scpca_filter == "Keep" ~ filter_levels[1],
+    scpca_filter == "Remove" & adt_scpca_filter == "Remove" ~ filter_levels[2],
+    scpca_filter == "Remove" & adt_scpca_filter == "Keep" ~ filter_levels[3],
+    scpca_filter == "Keep" & adt_scpca_filter == "Remove" ~ filter_levels[4],
+    TRUE ~ "Error"
+  ),
+  # make sure Keep is first 
+  filter_summary = forcats::fct_relevel(filter_summary, "Keep"),
+  keep_column = ifelse(
+    stringr::str_starts(filter_summary, "Keep"),
+    "Keep",
+    "Remove"
+  ))
+
+# faceted plot to show cells belonging to each filter group
+filtered_plot <- ggplot(filtered_coldata_df, aes(x = detected, y = subsets_mito_percent, color = keep_column)) +
+  geom_point(alpha = 0.5, size = 1) +
+  labs(
+    x = "Number of genes detected",
+    y = "Mitochondrial percentage"
+  ) +
+  facet_wrap(vars(filter_summary)) +
+  theme(
+    plot.title = element_text(hjust = 0.5),
+    legend.position = "none"
+  )
+
+# Top ADTs ---------------------------------------------------------------------
+
+# Calculate variance for each ADT
+adt_var <- altExp(processed_sce) |>
+  logcounts() |>
+  apply(1, var, na.rm = TRUE)
+
+# Get the top 4
+top_adts <- adt_var[order(adt_var, decreasing = TRUE)[1:4]] |>
+  names()
+
+# ADT expression density -------------------------------------------------------
+
+# grab expression for top ADTs from counts
+var_adt_exp_df <- logcounts(altExp(processed_sce))[top_adts, ] |>
+  # as.matrix needs to come _first_
```

Comment:
I too have learned this the hard way.

Reply:
😂 I believe this is your original comment since I did copy some of this code

Reply:
Past me did a good job!!